app/src/main/java/com/github/se/icebreakrr/ui/profile/ProfileEdit.kt

```diff
@@ -152,14 +160,28 @@ fun ProfileEditingScreen(
                 ProfilePictureSelector(
                     url = user.profilePictureUrl,
                     localBitmap = tempBitmap,
+                    pictureChangeState = pictureChangeState,
                     size = profilePictureSize,
                     onSelectionSuccess = { uri ->
                       profilesViewModel.generateTempProfilePictureBitmap(context, uri)
-                      isModified = true
+                      // save modification and implicitly sets isModified at next recomposition:
+                      profilesViewModel.saveEditedProfile(getEditedProfile())
                       navigationActions.navigateTo(Screen.CROP)
                     },
                     onSelectionFailure = {
                       Toast.makeText(context, "Failed to select image", Toast.LENGTH_SHORT).show()
+                    },
+                    onDeletion = {
+                      when (pictureChangeState) {
+                        UNCHANGED -> { // delete current profile picture
+                          if (user.profilePictureUrl != null) {
+                            profilesViewModel.setPictureChangeState(TO_DELETE)
+                            isModified = true
+                          }
+                        }
+                        // else cancel changes
+                        else -> profilesViewModel.setPictureChangeState(UNCHANGED)
```

Comment:
Question : Why using a "when" here is you have only two conditions : could you use just "if else" or is there an other reason behind this ?

Reply:
I always use the `when` syntax with enums as I find it more readable and more scalable (if I have to add more states to the enum, this code will be easy to update and still readable.)